src/core/auth/OidcSimpleClient.ts

```diff
@@ -92,22 +142,38 @@ export class OidcSimpleClient {
     try {
       const storedTokensJson = this.storage.getItem(LOCALSTORAGE_AUTH_KEY);
       if (storedTokensJson) {
-        const { token, refreshToken } = JSON.parse(storedTokensJson);
-        if (token && refreshToken) {
-          const decodedToken = parseToken(token);
-          // ensure timeToRefresh is shorter than tokenLifetime
-          this.timeToRefresh = Math.min(
-            Math.trunc((decodedToken.tokenLifetime * 1000) / 5),
-            TIME_TO_REFRESH_MS,
-          );
-          const decodedRefreshToken = parseToken(refreshToken);
-          this.setAuth(
-            token,
-            refreshToken,
-            decodedToken.expiringDate,
-            decodedRefreshToken.expiringDate,
-          );
-          return true;
+        const stored = JSON.parse(storedTokensJson);
+        if (stored.token && stored.refreshToken) {
+          try {
+            // Validate token format
+            const token = this.sanitizeToken(stored.token);
+            const refreshToken = this.sanitizeToken(stored.refreshToken);
+
+            const decodedToken = parseToken(token);
+            const decodedRefreshToken = parseToken(refreshToken);
```

Comment:
Feels like sanitizing + parsing can be extracted into a function, there's some code duplication right now (line 91)